pkg/source/source.go

```diff
@@ -4,24 +4,30 @@ import (
 	"fmt"
 )
 
-const doxIdFmt = "dox: %s"
-const doxIgnore = "dox: ignore"
-const doxOmitNotice = "dox: omit-notice"
+const doxHeaderFmt = "dox: %s"
+
+var doxDirectiveLiterals = map[string]string {
+	"ignore": "ignore",
+	"omitNotice": "omit-notice",
+	"rootPage": "root-page",
+}
```

Comment:
We might consider enumerating directives using constants if the number of directives continues to grow, which lets us reason about directives as integers, or even do bitwise operations. Check out https://splice.com/blog/iota-elegant-constants-golang/ if you're interested in what that could look like.